src/components/range-slider/range-slider.tsx

```diff
@@ -0,0 +1,130 @@
+import classNames from 'classnames';
+import styles from './range-slider.module.scss';
+
+interface RangeSlider {
+    startValue: number;
+    endValue: number;
+    onStartValueChange: (value: number) => void;
+    onEndValueChange: (value: number) => void;
+    /**
+     * The lowest permitted value.
+     */
+    minValue: number;
+    /**
+     * The highest permitted value.
+     */
+    maxValue: number;
+    /**
+     * The granularity that the values must adhere to. @default 1
+     */
+    step?: number;
+    /**
+     * Allows to format the displayed start and end values. For example, add a currency symbol,
+     * format with the specified number of decimal places, etc.
+     */
+    formatValue?: (value: number) => string;
+    startInputName?: string;
+    endInputName?: string;
+}
+
+/**
+ * A slider component for selecting a numeric range.
+ */
+export const RangeSlider = ({
+    startValue,
+    endValue,
+    onStartValueChange,
+    onEndValueChange,
+    minValue,
+    maxValue,
+    step,
+    formatValue = (value) => value.toString(),
+    startInputName,
+    endInputName,
+}: RangeSlider) => {
+    const handleStartValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newStartValue = Number(event.target.value);
+        if (newStartValue <= endValue) {
+            onStartValueChange(newStartValue);
+        }
+    };
+
+    const handleEndValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newEndValue = Number(event.target.value);
+        if (newEndValue >= startValue) {
```

Comment:
This should be `max` rather than `if`. Same for the `handleStartValueChange`.

https://github.com/user-attachments/assets/bfef75a0-d4cd-4349-af62-64dd3b1cc035